meson.build

```diff
@@ -69,12 +69,17 @@ openssl_dep = dependency('openssl',
                          fallback : ['openssl', 'libssl_dep'])
 if openssl_dep.found()
   conf.set('CONFIG_OPENSSL', true, description: 'Is OpenSSL available?')
+  # XXX Check the 3.x API for completeness; needed for LibreSSL
+  openssl_3_complete = (openssl_dep.version().version_compare('>=3.0.0') and
```

Comment:
Please don't add `XXX` to the comment? No idea what it suppose to mean. 

So this is a bit backwards in my opinion. LibreSSL is trying to be like OpenSSL, hence I prefer your idea with 

`openssl_is_libressl = cc.has_header_symbol('openssl/opensslv.h', 'LIBRESSL_VERSION_NUMBER')`

This explicitly tells us what is happening. We could rename CONFIG_OPENSSL_1 to something like `CONFIG_OPENSSL_API_V1` to make it even more explicit.